### ISSUE TITLE
[HOLD for payment 2023-03-20] [HOLD #12370][$2000] Audit forms and fix inconsistencies with focus, tab and shift + tab behavior 3/4

### ISSUE THREAD

**Issue by dylanexpensify** [Expensify/App#7917]
We should audit all our forms and fix any inconsistencies with focus, tab, shift + tab and enter behavior. The expected behavior is as follows:

1. `Tab` navigates to the next input.
2. `Shift + tab` navigates to the previous input.
3. `Enter` submits the form.
4. `Space` toggles checkboxes/dropdowns.

**Note:** We should make sure that tabbing cycles through the form in an order that makes sense, usually top to bottom.

Here's a list of forms to be audited:
- [ ] [IOU flow](https://github.com/Expensify/App/blob/30118b75ba9693ee5518464c8d7073a9c6883b19/src/components/IOUConfirmationList.js)
- [ ] [KYC flow](https://github.com/Expensify/App/blob/30118b75ba9693ee5518464c8d7073a9c6883b19/src/pages/EnablePayments/AdditionalDetailsStep.js)
- [ ] [New Room screen](https://github.com/Expensify/App/blob/main/src/pages/workspace/WorkspaceNewRoomPage.js)
- [ ] [Add Debit card flow](https://github.com/Expensify/App/blob/main/src/pages/settings/Payments/AddDebitCardPage.js)
- [ ] [New password form](https://github.com/Expensify/App/blob/30118b75ba9693ee5518464c8d7073a9c6883b19/src/pages/settings/NewPasswordForm.js)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @marcochavezf (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Christinadobrzyn (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by dylanexpensify** [comment]
Note: decision to split this main issue up into smaller issues came from [this convo](https://expensify.slack.com/archives/C01SKUP7QR0/p1645014281039819)

**Comment by marcochavezf** [comment]
I've been working on an N7 issue these last few days and still today, but I will come back to this issue today at EOD or tomorrow morning.

**Comment by Christinadobrzyn** [comment]
Followed up about the price for this Upwork job in Slack [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1646197102580479?thread_ts=1645014281.039819&cid=C01SKUP7QR0)

**Comment by marcochavezf** [comment]
Oh I hadn't seen the conversation. I think this case I can just unassign myself since the `Exported` label will be applied to assign reviewers.

**Comment by Christinadobrzyn** [comment]
Dylan confirmed this GH job is $250. Created Upwork job

Internal job posting - https://www.upwork.com/ab/applicants/1499638878787203072/job-details
External job positing - https://www.upwork.com/jobs/~014a2b9d0ae263c283

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @Santhosh-Sellavel (`Exported`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @thienlnam (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by mdneyazahmad** [comment]
## Proposal
These 4 related issues are very similar. I proposed the solution here https://github.com/Expensify/App/issues/7918#issuecomment-1063298280

**Comment by thienlnam** [comment]
Still waiting on a volunteer to audit the forms

**Comment by Christinadobrzyn** [comment]
Price increase to $500

**Comment by thienlnam** [comment]
Price increased to $1000, still looking for a volunteer

**Comment by Christinadobrzyn** [comment]
Job price increase to $2,000

**Comment by mdneyazahmad** [comment]
I have noticed that we have a team of accessibility engineer may be they can help us audit and the contributor can submit proposal for fixing that issues?

**Comment by OneDevStopShop** [comment]
Hi I am interested in this issue.  I am confused about the expectations, do we just need to update the tab functionality if it does not work as expected? 

How do you access the [KYC flow](https://github.com/Expensify/App/blob/30118b75ba9693ee5518464c8d7073a9c6883b19/src/pages/EnablePayments/AdditionalDetailsStep.js) page?

**Comment by thienlnam** [comment]
> I have noticed that we have a team of accessibility engineer may be they can help us audit and the contributor can submit proposal for fixing that issues?

Good suggestion, though our accessibility QA team is going through a different flow so they wouldn't test this unless a PR happened to touch these forms.  We've narrowed the scope of this issue and since it's just checking a few things I think a contributor should be able to do it 

@OneDevStopShop If you're looking for an example of what needs to be done,  feel free to check this other similar issue out https://github.com/Expensify/App/issues/7523

> How do you access the [KYC flow](https://github.com/Expensify/App/blob/30118b75ba9693ee5518464c8d7073a9c6883b19/src/pages/EnablePayments/AdditionalDetailsStep.js) page?
Good question, this step comes up when we're setting up your Expensify Wallet.  Check out the `EnablePaymentsPage` and you'll likely have to be on a new account.

**Comment by OneDevStopShop** [comment]
I looked through the issue you linked and would like to take this issue

**Comment by Christinadobrzyn** [comment]
Hired @OneDevStopShop for the job and @Santhosh-Sellavel as the C+ in Upwork!

**Comment by Santhosh-Sellavel** [comment]
@OneDevStopShop If you have any questions please feel to post them here. Thanks!

**Comment by mdneyazahmad** [comment]
I would like to take this one. I am also assigned other form audit issues and working on it. Thank you.

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad Any update?

**Comment by Christinadobrzyn** [comment]
Wait, sorry @thienlnam I might have missed something. 

I hired @OneDevStopShop for this job based on the 'thumbs up' to this post - https://github.com/Expensify/App/issues/7917#issuecomment-1090410859

Should I hire @mdneyazahmad instead?

**Comment by thienlnam** [comment]
@Christinadobrzyn No you were correct in hiring them. Though from this comment https://github.com/Expensify/App/issues/7917#issuecomment-1093009100 it seems like they're having an issue with Upwork and so I've hired @mdneyazahmad for the job instead

**Comment by Christinadobrzyn** [comment]
Oh I'm so sorry, not sure how I missed that comment. Thank you for clarifying! 

I hired @mdneyazahmad for the job! Sorry about the confusion.

**Comment by luacmartins** [comment]
@mdneyazahmad please note that `4. Space toggles checkboxes.` was added to the list of expected behavior.

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad Kindly provide an update on how is it coming up?

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel sorry for delay. I will work on this one today. Thank you.

**Comment by Christinadobrzyn** [comment]
I'm going to be ooo until April 26th so I'm going to assign a new CM to this issue. Thanks!

**Comment by mdneyazahmad** [comment]
This issue depends on https://github.com/Expensify/App/issues/7623 I will have to wait until that issue is fixed.

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad How does it depend on the #7623? 

cc: @thienlnam

**Comment by mdneyazahmad** [comment]
In this issue we have to fix inconsistency with `tab` and `shift + tab` which seems to work fine and also the form should submit when enter is pressed. 

#### Current implementation:

`Button` component accepts `enterOnPress` as prop and register a global keypress listener for `Enter` and calls form submit function. This however does not work as expected because enter can be pressed to press a button outside the form https://github.com/Expensify/App/issues/7918#issuecomment-1096631126

This depends on that issue because that tries to solve the same core problem how to handle the `Enter` key pressed and PR for this issue will depend on the decision taken there.

cc: @Santhosh-Sellavel

**Comment by Santhosh-Sellavel** [comment]
Thanks, @mdneyazahmad. Let's not wait for it.
Perform the audit on each form and list out the issues/inconsistencies.
 if there are issues with `Enter to submit`  feel free to point out that too.
 
 cc: @thienlnam

**Comment by mdneyazahmad** [comment]
### Form audit

#### IOU Confirmation Page
1. `tab` and `shift + tab` works as expected.
2. `enter` submits the form.
3. There is no checkbox.

https://user-images.githubusercontent.com/77761491/163789014-37959fa4-795a-41e4-be5c-eda96c64b392.mp4

#### New Room Screen
1. `tab` and `shift + tab` works as expected.
2. `enter` submits the form.
3. There is no checkbox.

https://user-images.githubusercontent.com/77761491/163789300-450bb9cd-b9a2-4cdc-a39a-8803e9cd2a07.mp4

#### Add Debit Card Flow
1. `tab` and `shift + tab` works as expected.
2. `enter` submits the form.
3. `space` does not toggle checkbox.

https://user-images.githubusercontent.com/77761491/163789610-7fe7cde7-eaa6-4408-9937-758217b04a9d.mp4

#### New Password Form
1. `tab` and `shift + tab` works as expected.
2. `enter` submits the form.
3.  `space` toggles checkbox (`eye icon`).

https://user-images.githubusercontent.com/77761491/163789733-b9d4b2de-9d49-4a9d-9159-74d0e6b2cdf7.mp4

**Comment by mdneyazahmad** [comment]
There are some other issues.

#### New Room Screen
1. Option select focus styles is applied even when it goes out of focus.

#### Add Debit Card Flow
1. Option select (State Field) focus styles is applied even when it goes out of focus.
2. State Field does not have error text.
3. It automatically enters some text (possibly first 5 letters of email) into zip code field when page loads.

#### New Password Form
1. Submit button is disabled by default.
2. When we enter invalid password and press enter to submit. It produces error and validates the password on input. It should not validate on input instead it should on submit.
3. It does not have red outline when it is invalid.

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel Could you help me navigate `KYC Flow`?

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad 

Navigate to `/settings/payments/enable-payments` which KYCFLow start point

These are steps in the flow, make some code tweaks to test the same.

https://github.com/Expensify/App/blob/6c53be1ea7a022e8a5e788cc7e7efd62c4573474/src/pages/EnablePayments/EnablePaymentsPage.js#L59-L64

@luacmartins can you confirm this.

**Comment by luacmartins** [comment]
@Santhosh-Sellavel correct! I'd just hardcode each step individually to test each of them.

**Comment by mdneyazahmad** [comment]
Sometimes, it depends on the data from previous step and if not present there it will crash. I will test and let you know if this method works. Thank you!

**Comment by mdneyazahmad** [comment]
#### KYC Flow

##### Step 1
1. `tab` and `shift + tab` works as expected.
2. `enter` submits the form.
3. There is no checkbox.

https://user-images.githubusercontent.com/77761491/164278074-db37063d-92e9-4801-a54f-bfe1bc2efb90.mp4



##### Step 2
There is no form element.
<img width="374" alt="Screenshot 2022-04-20 at 9 50 41 PM" src="https://user-images.githubusercontent.com/77761491/164277986-5c9101a4-320a-4849-b3f6-22ff6a797c43.png">


##### Step 3
1. `tab` and `shift + tab` works does not work as expected. `View printer friendly` is skipped.
2. `enter` does not submits the form.
3. `space` does not toggle checkboxes.

https://user-images.githubusercontent.com/77761491/164278088-faab2d7c-68ec-4dc5-bc5c-ce8f28886b09.mp4



##### Step 4
There is no form element. It has just text.

**Comment by mdneyazahmad** [comment]
### Other issues

1. `Step 1` and `step 3` have footer button that is not fixed at the bottom.

https://user-images.githubusercontent.com/77761491/164279347-288eec03-1777-4cb5-98f5-b9ff8955c632.mp4

2. `Step 2` footer button has double spacing.
<img width="373" alt="Screenshot 2022-04-20 at 10 03 44 PM" src="https://user-images.githubusercontent.com/77761491/164279649-3cd4a40e-6c4d-475c-bf01-66837dced87d.png">

**Comment by thienlnam** [comment]
@mdneyazahmad Thanks for the videos

For the KYC flow -
Step 1 and step 2 look good.  No changes needed there

Let's address these issues in step 3
> tab and shift + tab works does not work as expected. View printer friendly is skipped.
enter does not submits the form.
space does not toggle checkboxes.

As for the other issues, let's go ahead and have those addressed in the PR as well

**Comment by mdneyazahmad** [comment]
I am not sure about the checkbox implementation. Current version uses a button for checkbox which is toggled when button is pressed `(Enter)`.  The ideal component to use here is the native checkbox element (eg. `<input type="checkbox">` on web). This way browser will provide default behaviour for accessibility and will dispatch an `SubmitEvent` when `Enter` is pressed.

I am curious to know how the other issue is solved and based on that the implementation may change here for form submit with `Enter`. 

cc: @thienlnam @Santhosh-Sellavel

**Comment by Santhosh-Sellavel** [comment]
> I am curious to know how the other issue is solved and based on that the implementation may change here for form submit with Enter.

Sorry What other issue? could you add more context, please!

@mdneyazahmad

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad 

Regarding IOU Confirmation we should check all IOU operation confirmation pages.
1. Request Money
2. Send Money
3. Split Bill Using FAB in LHN
4. Split Bill from group chat/multiple participants.

Thanks!

**Comment by thienlnam** [comment]
> The ideal component to use here is the native checkbox element (eg. <input type="checkbox"> on web)

I think this is fine, not sure why we would need it to be a button

**Comment by thienlnam** [comment]
> The ideal component to use here is the native checkbox element (eg. <input type="checkbox"> on web).

This is interesting, we could add a web version of the checkbox that just uses this but is there a way we can have this done with React Native adding a shortcut when the checkbox is focused?

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel I was talking about https://github.com/Expensify/App/issues/7623 (other issue). I am sorry I missed that to audit with all four pages posting soon. Thank you!

**Comment by mdneyazahmad** [comment]
I am ooo and will be back in 1 -2 days.
Remaining task
- [ ] audit https://github.com/Expensify/App/issues/7917#issuecomment-1106834209
- [ ] fix checkbox

Thank you!

**Comment by mdneyazahmad** [comment]
##### Request money
1. `tab` and `shift + tab` works as expected.
2. `enter` submits the form.
3. There is no checkbox.
4. `enter` toggles selections (OptionsList) as well as navigates the user to the next step.

https://user-images.githubusercontent.com/77761491/166134375-6db1fcc1-99ce-44df-b683-5e4bb2c8daa8.mp4

##### Send money
1. `tab` and `shift + tab` works as expected.
2. `enter` submits the form.
3. There is no checkbox.
4. `enter` toggles selections (OptionsList) as well as navigates the user to the next step.

https://user-images.githubusercontent.com/77761491/166134376-d193fb1c-4bad-49b7-a45d-b36e2ec2e154.mp4

##### Split from fab
1. `tab` and `shift + tab` works as expected.
2. `enter` submits the form.
3. There is no checkbox.
4. `enter` toggles selections and you have to press `tab` multiple times to focus on the submit button and then press `enter` to navigate to the next step.

https://user-images.githubusercontent.com/77761491/166134379-ee1fd82c-60d5-4f0e-8ee0-7ae16022c786.mp4

##### Split from group
1. `tab` and `shift + tab` works as expected.
2. `enter` submits the form.
3. There is no checkbox.
4. `enter` toggles selections (OptionsList) as well as submits the form.

https://user-images.githubusercontent.com/77761491/166134381-8af709a5-61d0-4eca-b509-92ca770ac6b1.mp4

**Comment by mdneyazahmad** [comment]
This is my initial attempt to fix the checkbox accessibility.
1. Created a separate implementation for `native` and others.
2. `native` implementation is exactly same.
3.  `web` and` desktop` added custom logic to handle `onClick` and `onKeyDown` to listen for `space`.

You can check the diff here https://github.com/Expensify/App/compare/main...mdneyazahmad:fix/checkbox

**Comment by mdneyazahmad** [comment]
There are issues with `OptionsList`. It toggles selection with `enter` and when you press it toggles as well as submits the form that is not the user is expecting. We are discussing the accessibility of OptionsList in the other issue may be continue the discussion here https://github.com/Expensify/App/issues/7916#issuecomment-1113592130

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel PR will be ready for review in few hours.

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel PR is ready for review https://github.com/Expensify/App/pull/8874

**Comment by Santhosh-Sellavel** [comment]
I see only check box changes there what about the issues mentioned here, https://github.com/Expensify/App/issues/7917#issuecomment-1101323703

**Comment by stephanieelliott** [comment]
^^ Bump @mdneyazahmad, are you working on including these in the PR?

**Comment by mdneyazahmad** [comment]
I am sorry for the delay PR is almost ready.

**Comment by Santhosh-Sellavel** [comment]
Yes, group upto maximum two tests in a single video.

Break down into seperate comments for each test or grouped tests, yes upload all platform videos thanks!

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel Could you please review it (#8874) and if looks good merge it. I am trying to create a separate PR to fix `tab` and `enter` inconsistency and the checkbox PR should me merged for testing.

I created a separate PR for the checkbox because that is common to other audit issues.

**Comment by mdneyazahmad** [comment]
I am trying to handle other issues in a separate PR. Checkbox issue is common to the other issue I am working therefore I created a separate PR for that. 

PR for other issue is drafted here https://github.com/Expensify/App/pull/8747 but I am introducing one issue if we go with the `pressOnEnter` on Button  https://github.com/Expensify/App/issues/7918#issuecomment-1134216565

cc: @Santhosh-Sellavel

**Comment by Santhosh-Sellavel** [comment]
Thanks for the update, I'll review that alone by tomorrow!

**Comment by Santhosh-Sellavel** [comment]
Just waiting for an update from @mdneyazahmad regarding issue fixes for this audit.

Also, checkbox PR is currently in review

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel We have an issue with Button pressOnEnter  https://github.com/Expensify/App/issues/7918#issuecomment-1144280328. Asked on slack here https://expensify.slack.com/archives/C01GTK53T8Q/p1654195847140899

**Comment by stephanieelliott** [comment]
Hey @mdneyazahmad any update on this? There was a suggestion on the Slack convo you'd linked, will it solve this issue? 

> The key to solving this problem generally is the [shouldBubble](https://github.com/Expensify/App/blob/719f18405b4d902bbea6435d7f9b707c6f2d13f5/src/libs/KeyboardShortcut/index.js#L143) configuration parameter of keyboard event handlers.

**Comment by mdneyazahmad** [comment]
@stephanieelliott Sorry for the late reply. I tested that solution does not work.

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad What's the current status of this issue, we have been on this for months. Can we speed this up?

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel I do want to finish this asap. But we are stuck at handling ENTER for submitting the form as it creates secondary interaction. Could you please join discussion [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1654195847140899), [and here](https://expensify.slack.com/archives/C01GTK53T8Q/p1655778876155109) and help move forward.

Related PR: 
1. https://github.com/Expensify/App/pull/8747
2. https://github.com/Expensify/App/pull/8874

**Comment by stephanieelliott** [comment]
Discussions are ongoing in Slack...

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad  We have an update on the slack thread can you check it out.

**Comment by stephanieelliott** [comment]
Hey @dylanexpensify, reassigning this one since I am about to head OOO. To save you some scrolling, we got bit stuck on the best way to handle button behavior, discussion on that is going on here: https://expensify.slack.com/archives/C01GTK53T8Q/p1654195847140899. Thank you!

**Comment by dylanexpensify** [comment]
Sounds good! Thanks @stephanieelliott and have a great OOO time!

**Comment by thienlnam** [comment]
@mdneyazahmad Any update here?

**Comment by parasharrajat** [comment]
@mdneyazahmad Any update? I would suggest pushing things to reach the finish line. If there are no regular updates, we will close the PR and reevaluate the current state of the issue.

**Comment by mdneyazahmad** [comment]
@parasharrajat I am sorry for the delay I was sick in the last few days. I think we are not yet reached the conclusion [slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1654195847140899).

**Comment by dylanexpensify** [comment]
Any updates here? cc @mdneyazahmad

**Comment by mdneyazahmad** [comment]
@dylanexpensify Still discussing. I will post some new ideas.

**Comment by thienlnam** [comment]
@mdneyazahmad Looks like there has been consensus with 1, are we able to move forward with that now?

> Continue using a single global keyboard event handler library, patching it with the solution from [this PR](https://github.com/Expensify/App/pull/9749) so that local event handlers can override the global ones

**Comment by mdneyazahmad** [comment]
@thienlnam I had posted my thoughts on that method. I will re review it and post the conclusion. Thank you!

**Comment by mdneyazahmad** [comment]
Not overdue. @Santhosh-Sellavel Could you please check it? I propose an alternate solution that may fix the issue https://github.com/Expensify/App/pull/9749#discussion_r946663417

**Comment by Santhosh-Sellavel** [comment]
@thienlnam this issue is not fully completed can we update the title to avoid any confusions
2 more PRs to go!

cc: @dylanexpensify

**Comment by thienlnam** [comment]
Pending PR

**Comment by thienlnam** [comment]
Looks like this PR has been pending review for a while https://github.com/Expensify/App/pull/10965#issuecomment-1261256955

Bump @parasharrajat

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad whats the status here?

**Comment by parasharrajat** [comment]
Under review now

**Comment by dylanexpensify** [comment]
@parasharrajat @mdneyazahmad  any updates?

**Comment by mdneyazahmad** [comment]
Under review https://github.com/Expensify/App/pull/10965

**Comment by dylanexpensify** [comment]
Thanks @mdneyazahmad !

**Comment by dylanexpensify** [comment]
any updates @mdneyazahmad

**Comment by dylanexpensify** [comment]
On hold!

**Comment by dylanexpensify** [comment]
on hold

**Comment by dylanexpensify** [comment]
on hold

**Comment by luacmartins** [comment]
@dylanexpensify why do we have a hold on this issue?

**Comment by mdneyazahmad** [comment]
@luacmartins a component is being introduced in the PR (https://github.com/Expensify/App/pull/12370) that will be used here. Therefore, this is on hold. Thanks!

**Comment by puneetlath** [comment]
Has the New Room screen already been audited? Because I've been updating that to use the form component as part of https://github.com/Expensify/App/issues/12678.

**Comment by dylanexpensify** [comment]
@luacmartins unsure - I see @puneetlath added it on hold, any insight Puneet? Or are we good to move forward?

**Comment by trjExpensify** [comment]
Let's discuss this one in Slack [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1670867974272229), as I think this should be held on WAQ as it's an issue related to keyboard navigation accessibility.

**Comment by Santhosh-Sellavel** [comment]
We have why this is on hold answered [here](https://github.com/Expensify/App/issues/7917#issuecomment-1335544236)

**Comment by thienlnam** [comment]
Still on hold for this PR here - https://github.com/Expensify/App/pull/12370

**Comment by JmillsExpensify** [comment]
Commented similarly in a related issue, but two things:

1. WAQ is over, so let's take this issue off hold. 
2. Now that this is off hold, I think we have a decision before us. Do we do anything at all? I actually vote no, we shouldn't. I still don't think keyboard navigation is a priority – including but not limited to tab, shift + tab, etc. – so better to close this issue and move on. Please let me know if I'm missing something!

**Comment by luacmartins** [comment]
I think we are really close to merging the [PR](https://github.com/Expensify/App/pull/12370), so I think we can continue with this one.

**Comment by luacmartins** [comment]
Oh nvm my comment above, that PR is solving a different [issue](https://github.com/Expensify/App/pull/12370). @thienlnam @mdneyazahmad @Santhosh-Sellavel what's the latest on this one?

**Comment by Santhosh-Sellavel** [comment]
> @luacmartins a component is being introduced in the PR (#12370) that will be used here. Therefore, this is on hold. Thanks!

This is on hold for the same PR

**Comment by JmillsExpensify** [comment]
Ah, if this issue is on hold for a specific issue, then let's add it. I mainly got here combing the repo for issues held on WAQ. All that said, I think this is a really low priority issue.

**Comment by mdneyazahmad** [comment]
Update: PR (https://github.com/Expensify/App/pull/12370) is merged! I will create the PR for this issue this weekend. Thanks!

**Comment by mdneyazahmad** [comment]
Out of these components:

1. [IOU flow](https://github.com/Expensify/App/blob/30118b75ba9693ee5518464c8d7073a9c6883b19/src/components/IOUConfirmationList.js)
2. [KYC flow](https://github.com/Expensify/App/blob/30118b75ba9693ee5518464c8d7073a9c6883b19/src/pages/EnablePayments/AdditionalDetailsStep.js)
3. [New Room screen](https://github.com/Expensify/App/blob/main/src/pages/workspace/WorkspaceNewRoomPage.js)
4. [Add Debit card flow](https://github.com/Expensify/App/blob/main/src/pages/settings/Payments/AddDebitCardPage.js)
5. [New password form](https://github.com/Expensify/App/blob/30118b75ba9693ee5518464c8d7073a9c6883b19/src/pages/settings/NewPasswordForm.js)


After this PR (https://github.com/Expensify/App/pull/12370) and this (https://github.com/Expensify/App/pull/8874), the issue seems to be fixed as I have made the changes in `Form` component to handle the form submit action with enter key, and these components import and use `Form`. Thus the issue is fixed.

[IOU flow]: Works fine
[KYC flow]: I can not navigate to this, but I have checked the code of this components. This also seems to be fixed as this also uses the updated `Form` component.
[New Room screen]: Works fine
[Add Debit card flow]: Works fine
[New password form]: This is fixed in this PR (https://github.com/Expensify/App/pull/12370)

This is the change I am referring to

https://github.com/Expensify/App/blob/142fbba48c2d5b17c809eff8f85cfc3611dee47b/src/components/Form.js#L282

cc: @luacmartins @Santhosh-Sellavel

**Comment by Santhosh-Sellavel** [comment]
Cool, we had few other issues other than submit issue right? Or all the issues are fixed?

Whats next @mdneyazahmad?

**Comment by Santhosh-Sellavel** [comment]
Tomorrow I'll go through the thread to look if anything is missed and confirm everything is working.

**Comment by Santhosh-Sellavel** [comment]
> Cool, we had few other issues other than submit issue right? Or all the issues are fixed?
> 
> Whats next @mdneyazahmad?
> 
> 

@mdneyazahmad bump!

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad 

> [KYC flow]: I can not navigate to this, but I have checked the code of this components. This also seems to be fixed as this also uses the updated Form component.

[Here](https://github.com/Expensify/App/issues/7917#issuecomment-1101481048) are the steps to navigate this flow.

Okay, I've gone through the thread, next steps would be. Since it's been a while we just need to re-run the audit on the [flow/pages ](https://github.com/Expensify/App/issues/7917#issue-1150479233) again please share screen records of the same i.e. if all looks good we can confirm and close this one. Thanks


Cc: @thienlnam

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel thanks. I will update sometime today.

**Comment by mdneyazahmad** [comment]
IOU


https://user-images.githubusercontent.com/77761491/216594629-a7a24703-7cb2-4b5f-8921-c928b3a37c46.mp4

https://user-images.githubusercontent.com/77761491/216594786-4034c1f7-62a1-467e-a365-45d1f55fbb08.mp4

https://user-images.githubusercontent.com/77761491/216596073-187239bf-1d58-4fc6-8061-37eb8146e446.mp4

KYC


https://user-images.githubusercontent.com/77761491/216594666-fe8a884c-6cd8-4e7f-b542-0baa9dd3fc66.mp4

New Room


https://user-images.githubusercontent.com/77761491/216594693-2ece72e1-85f0-4a67-886b-a7c74a828aee.mp4

Add debit card


https://user-images.githubusercontent.com/77761491/216594724-50d143d9-d4cf-4640-9271-cd4271ce7100.mp4

https://user-images.githubusercontent.com/77761491/216594906-05d2f488-43d7-435f-aa0d-1080ed7d599c.mp4


Set password

https://user-images.githubusercontent.com/77761491/216594741-c97ec38b-d969-4642-bcad-a863ac8f8973.mp4

cc: @Santhosh-Sellavel

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel this works fine. Let me know if you find any issues.

**Comment by Santhosh-Sellavel** [comment]
Sorry for the inconvenience, @mdneyazahmad seems accidentally missed this one. I'll update by Monday or sooner, seems most likely to do nothing here. I'll test and update.

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad 

Weird shouldn't the `tab` focus to the `Next` button?

https://user-images.githubusercontent.com/85645967/218585018-8b4ba81c-eab3-4472-b56d-66e9498e970c.mov

**Comment by Santhosh-Sellavel** [comment]
Ah, I just noticed the Next button is disabled 😄. @luacmartins should still keep this Next button disabled?

**Comment by Santhosh-Sellavel** [comment]
Bug:
IOU input amount should not be selected on focus using

https://user-images.githubusercontent.com/85645967/218586011-4f939eba-5416-4e26-9d10-bd57316a4070.mov

Focus is not visible on drop down

https://user-images.githubusercontent.com/85645967/218586733-83f6bde1-d462-4438-bf9f-613bc9cfbc43.mov

**Comment by Santhosh-Sellavel** [comment]
What's your thought on the other issue, @luacmartins?

**Comment by Santhosh-Sellavel** [comment]
Will continue testing other tomorrow!

**Comment by luacmartins** [comment]
>IOU input amount should not be selected on focus using

This seems to be the standard behavior for inputs with values. 

https://user-images.githubusercontent.com/22219519/218737361-be6921c6-0c45-499a-881a-c40f39eecca2.mov

>Focus is not visible on the drop down error

I'm not sure that I understand this one - the focus seems to shift from one input to the other correctly. Could you please clarify?

**Comment by Santhosh-Sellavel** [comment]
> > Focus is not visible on the drop down error

> I'm not sure that I understand this one - the focus seems to shift from one input to the other correctly. Could you please clarify?

Yes it seems to be shifting, but not visible when the input (drop down) is in an error state

**Comment by Santhosh-Sellavel** [comment]
### KYC
https://user-images.githubusercontent.com/85645967/218895034-b815e348-75ed-4128-a311-8cb0908aa0ac.mov

### Debit
https://user-images.githubusercontent.com/85645967/218895012-845c1f93-35fb-4d36-96e5-a7a8e529517d.mov

**Comment by Santhosh-Sellavel** [comment]
## New password form

Pressing Enter should error instead goes to home page

https://user-images.githubusercontent.com/85645967/218895656-69cbb830-ace1-4b45-9504-9532aaa8c598.mov

**Comment by Santhosh-Sellavel** [comment]
@luacmartins

1. 
> > IOU input amount should not be selected on focus using
> 
> This seems to be the standard behavior for inputs with values.

We are good then!

2. Also  any update on [this](https://github.com/Expensify/App/issues/7917#issuecomment-1430424211)

3. This is a bug right https://github.com/Expensify/App/issues/7917#issuecomment-1430589582?

**Comment by luacmartins** [comment]
>Focus is not visible on the drop down error

Got it! Hmm that seems to be the case in the pages that use Form too. So if we were to fix it we would have to fix the Form component itself, which I think is outside the scope for this issue.

>This is a bug right https://github.com/Expensify/App/issues/7917#issuecomment-1430589582?

Seems like it, although I can't reproduce. Would you mind listing the steps?

**Comment by Santhosh-Sellavel** [comment]
@luacmartins 

 - Signup a new account using email
- Open magic received from mail
- Enter invalid password or don't type anything. Now Press Enter 

Pressing Enter should error, instead goes to home page

**Comment by luacmartins** [comment]
Thanks! I was able to reproduce and yes, that's a bug!

**Comment by JmillsExpensify** [comment]
Do we have a new issue for that one yet? If not, we can get the BugZero person assigned to create one. cc @dylanexpensify

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel Can you please confirm that [this](https://github.com/Expensify/App/issues/7917#issuecomment-1430589582) issue happens on PR, as I can see that when you open the link it redirects to `https://new.expensify.com/setpassword/uuid`.

You can follow this steps:
1. Signup a new account using email
2. Copy the link received in email.
3. Replace new.expensify.com as  `https://new.expensify.com/setpassword/uuid` -> `http://localhost/setpassword/uuid`
4. Now test

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad 
Ah thanks, I believe this [one](https://github.com/Expensify/App/issues/7917#issuecomment-1430589582) comes with in the scope of this audit. We should fix it here.

cc: @luacmartins @thienlnam

**Comment by luacmartins** [comment]
I agree. That's within the audit scope.

**Comment by mdneyazahmad** [comment]
I am sorry for being late here. I was ooo. @Santhosh-Sellavel I have drafted the PR. I need some help with the test in native device and desktop. How can open the url in those platform?

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel I have marked that PR ready for review, and posted in slack channel for help [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1677061465199609) about opening the url in desktop and native devices.

**Comment by MelvinBot** [comment]
📣 @mdneyazahmad! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by mdneyazahmad** [comment]
Contributor details
Your Expensify account email: ahamadalam66@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~011922ae9ff280b160

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.82-4 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15240


If no regressions arise, payment will be issued on 2023-03-20. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@Santhosh-Sellavel / @thienlnam] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@Santhosh-Sellavel / @thienlnam] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@Santhosh-Sellavel / @thienlnam] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@dylanexpensify] Determine if we should create a regression test for the bug.
- [ ] [@Santhosh-Sellavel] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to the [appropriate location](https://github.com/Expensify/App/blob/main/contributingGuides/TestRail_Navigation_C%2B.md) to ensure the same bug will not reach production again.
- [ ] [@dylanexpensify] Review the proposed regression test steps and location.
- [ ] [@dylanexpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by Santhosh-Sellavel** [comment]
I think we don't need a BZ Checklist. It's all N/A for this one!

@dylanexpensify 

cc: @thienlnam

**Comment by dylanexpensify** [comment]
@thienlnam can you confirm payout amounts of $2000? As well, is there a time bonus here?

**Comment by thienlnam** [comment]
Looks correct, and I believe this issue was created way back before we even added time bonuses so let's just do normal payouts for this issue

**Comment by dylanexpensify** [comment]
@Santhosh-Sellavel @mdneyazahmad please apply [here](https://www.upwork.com/jobs/~01238b5366b1965b7c)!

**Comment by Santhosh-Sellavel** [comment]
Done, thanks!

**Comment by dylanexpensify** [comment]
offers sent!

**Comment by Santhosh-Sellavel** [comment]
Accepted thanks @dylanexpensify!

**Comment by dylanexpensify** [comment]
Alright, we're all sorted here! Thanks for playing!
